src/init.cpp

```diff
@@ -1400,10 +1402,14 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
     for (bool fLoaded = false; !fLoaded && !ShutdownRequested();) {
         node.mempool = std::make_unique<CTxMemPool>(mempool_opts);
 
-        const ChainstateManager::Options chainman_opts{
+        ChainstateManager::Options chainman_opts{
             .chainparams = chainparams,
             .adjusted_time_callback = GetAdjustedTime,
+            .datadir = args.GetDataDirNet(),
         };
+        node::ReadDatabaseArgs(args, chainman_opts.block_tree_db);
+        node::ReadDatabaseArgs(args, chainman_opts.coins_db);
+        node::ReadCoinsViewArgs(args, chainman_opts.coins_view);
```

Comment:
I'd say we should remove LOC in init.cpp, not add them. What about moving this into a new function called `ApplyArgsManOptions`?

Also, no need to re-parse this in the loop? Could move outside like fae180c7fd85ef8a1aeb489691779c215cb9d5a2